SimTracker/Common/interface/TrackingParticleSelector.h

```diff
@@ -64,64 +64,94 @@ class TrackingParticleSelector {
     }
   }
 
+  bool isCharged(const TrackingParticle *tp) const { return (tp->charge() == 0 ? false : true); }
+
+  bool isInTime(const TrackingParticle *tp) const { return (tp->eventId().bunchCrossing() == 0); }
+
+  bool isSignal(const TrackingParticle *tp) const {
+    return (tp->eventId().bunchCrossing() == 0 && tp->eventId().event() == 0);
+  }
+
+  bool isStable(const TrackingParticle *tp) const {
+    for (TrackingParticle::genp_iterator j = tp->genParticle_begin(); j != tp->genParticle_end(); ++j) {
+      if (j->get() == nullptr || j->get()->status() != 1) {
+        return false;
+      }
+    }
+    // test for remaining unstabled due to lack of genparticle pointer
+    auto pdgid = tp->pdgId();
```

Comment:
use int pdgid = std::abs(tp->pdgId()); and retmove std::abs below

Reply:
Fixed